### PR TITLE
One build to rule them all

### PR DESCRIPTION
**Pull request by erikwijmans**
## Motivation and Context

We currently have two different wrappers to the cmake build system.  One driven by `build.sh` and one driven by `setup.py`.  Unfortunately, these two different ways to drive cmake output the bindings to different locations and work slightly differently.  This leads to confusion and weird errors from collisions between the two methods.

This PR seeks to resolve this issue by standardizing on `setup.py` as the only way to drive cmake and build things.  `build.sh` just becomes a dumb wrapper around `python setup.py build_ext --inplace`.  This works largely the same as before, the one difference being that the library for bindings is put at `habitat_sim/_ext` instead of `build/esp/bindings`.

## How Has This Been Tested

Built on various systems with both `build.sh` and `python setup.py install`.   Both methods result in a usable install

Note that `pip install .` also works in place of `python setup.py install`, but we can't direct people to use this as it doesn't expose the viewer.

## Types of changes

- [x] New feature (non-breaking change which adds functionality)

### CODE REVIEW

**Comment by dhruvbatra** [comment]
This is awesome!

File: habitat_sim/bindings/__init__.py [hunk 1]
```diff
@@ -4,18 +4,28 @@
...
+    "MultiGoalActionSpaceShortestPath",
+    "ActionSpacePathLocation",
+    "AttachedObject",
+    "AttachedObjectType",
+    "PinholeCamera",
+    "Sensor",
+    "SceneNode",
+    "SceneGraph",
+    "geo",
+]
```
[the remark this answers is not in the document]
Comment (erikwijmans) [reply habitat_sim/bindings/__init__.py]: Done

File: requirements.txt [hunk 2]
```diff
@@ -3,3 +3,4 @@ numpy
 pillow
 numpy-quaternion
 attrs
+numba
```
[the remark this answers is not in the document]
Comment (erikwijmans) [reply requirements.txt]: Done

File: setup.py [hunk 3]
```diff
@@ -11,13 +11,62 @@
 import os
 import os.path as osp
 import sys
-import platform
 import subprocess
 import builtins
+import re
```
[the remark this answers is not in the document]
Comment (erikwijmans) [reply setup.py]: Done for all imports (ran isort)

**Pull request merged**